pyuvdata/calfits.py

```diff
@@ -127,68 +115,98 @@ def write_calfits(self, filename, run_check=True,
                                           self.quality_array[:, :, :, :, :, np.newaxis]],
                                          axis=-1)
             else:
-                prihdr['CRVAL1'] = (4, 'Number of image arrays.')
                 pridata = np.concatenate([self.gain_array.real[:, :, :, :, :, np.newaxis],
                                           self.gain_array.imag[:, :, :, :, :, np.newaxis],
                                           self.flag_array[:, :, :, :, :, np.newaxis],
                                           self.quality_array[:, :, :, :, :, np.newaxis]],
                                          axis=-1)
 
-            if self.total_quality_array is not None:
-                sechdr['CTYPE1'] = ('Narrays', 'Number of total quality arrays.')
-                sechdr['CUNIT1'] = ('Integar', 'Number of total quality arrays. Increment.')
-                sechdr['CDELT1'] = 1
-                sechdr['CRVAL1'] = (1, 'Number of image arrays.')
-                secdata = self.total_quality_array[:, :, :, :, np.newaxis]
+        if self.total_quality_array is not None:
+            # Set headers for the hdu containing the total_quality_array
+            totqualhdr['CTYPE1'] = ('JONES', 'Jones matrix array')
+            totqualhdr['CUNIT1'] = ('Integer', 'representative integer for polarization.')
+            totqualhdr['CRPIX1'] = 1
+            totqualhdr['CRVAL1'] = self.jones_array[0]  # always start with first jones.
+            if self.Njones > 1:
+                totqualhdr['CDELT1'] = jones_spacing[0]
+            else:
+                totqualhdr['CDELT1'] = -1
+
+            totqualhdr['CTYPE2'] = ('TIME', 'Time axis.')
+            totqualhdr['CUNIT2'] = ('JD', 'Time in julian date format')
+            totqualhdr['CRPIX2'] = 1
+            totqualhdr['CRVAL2'] = self.time_array[0]
+            totqualhdr['CDELT2'] = time_spacing
+
+            totqualhdr['CTYPE3'] = ('FREQS', 'Valid frequencies to apply delay.')
+            totqualhdr['CUNIT3'] = 'Hz'
+            totqualhdr['CRPIX3'] = 1
+            totqualhdr['CRVAL3'] = self.freq_array[0][0]
+            totqualhdr['CDELT3'] = freq_spacing
+
+            # Nspws axis: number of spectral windows
+            totqualhdr['CTYPE4'] = ('NSPWS', 'Number of spectral windows.')
+            totqualhdr['CUNIT4'] = 'Integer'
+            totqualhdr['CRPIX4'] = 1
+            totqualhdr['CRVAL4'] = 1
+            totqualhdr['CDELT4'] = 1
+            totqualdata = self.total_quality_array
 
         if self.cal_type == 'delay':
             # Set header variable for delay.
 
             # ANTAXIS axis number differs between delay and gain because there's no frequency axis in delay
             prihdr['CTYPE5'] = ('ANTAXIS', 'See ANTARR in ANTENNA extension for values.')
             prihdr['CUNIT5'] = 'Integer'
-            prihdr['CRVAL5'] = 0
+            prihdr['CRVAL5'] = 1
```

Comment:
Checking that led me to some other minor fixes, it turns out that calfits wasn't using the axis length parameters at all and instead inserting the length parameters as their own keywords. I fixed that as well.